lib/bowlingScoreCard.js

```diff
@@ -0,0 +1,62 @@
+class BowlingScoreCard {
+  constructor() {
+    this.eachRollScoreArray = [];
+  }
+
+  roll = (pinDown) => {
+    this.eachRollScoreArray.push(pinDown);
+  };
+
+  overallScore = () => {
+    let score = 0;
+    let rollNumber = 0;
+    for (let i = 0; i < 10; i++) {
+      if (this.isASpareFrame(rollNumber)) {
+        score += 10 + this.spareBonusScore(rollNumber);
+        rollNumber += 2;
+      } else if (this.isAStrikeFrame(rollNumber)) {
+        score += 10 + this.strikeBonusScore(rollNumber);
+        rollNumber += 1;
+      } else {
+        score += this.sumPinsDownInFrame(rollNumber);
+        rollNumber += 2;
+      }
+    }
+    if (this.isASpareFrame(rollNumber - 2)) {
+      score += this.eachRollScoreArray[rollNumber];
+    }
+    return score;
+  };
+
```

Comment:
I understand the means by which you are calculating spare and strike bonuses and it clearly works. However, 
I find that relying on array indexing so heavily will often lead to bugs and complicates building on the code.

Reply:
Having looked at your code and use of a linked list, I will be looking into this further as an alternative to array indexing in the future. Thank you